web/client/plugins/Details.jsx

```diff
@@ -6,48 +6,118 @@
+import { mapInfoSelector, mapIdSelector, mapInfoDetailsUriFromIdSelector } from '../selectors/map';
+import { mapLayoutValuesSelector } from '../selectors/maplayout';
+import { isLoggedIn } from '../selectors/security';
+
+import Message from '../components/I18N/Message';
+import Details from '../components/details/Details';
+import { createPlugin } from '../utils/PluginsUtils';
+
+import details from '../reducers/details';
+import * as epics from '../epics/details';
 
 /**
  * Details plugin used for fetching details of the map
  * @class
  * @memberof plugins
  */
 
-module.exports = {
-    DetailsPlugin: connect((state) => ({
-        active: get(state, "controls.details.enabled"),
+export default createPlugin('Details', {
+    component: connect((state) => ({
+        show: get(state, "controls.details.enabled"),
+        loading: loadingSelector(state),
+        loadFlags: loadFlagsSelector(state),
+        mapInfo: mapInfoSelector(state),
+        canEdit: isLoggedIn(state) && get(mapInfoSelector(state), 'canEdit'),
+        settings: settingsSelector(state),
         map: mapFromIdSelector(state, mapIdSelector(state)),
-        dockStyle: mapLayoutValuesSelector(state, {height: true}),
-        detailsText: currentMapDetailsTextSelector(state)
+        content: contentSelector(state),
+        editedContent: editedContentSelector(state),
+        contentChanged: contentChangedSelector(state),
+        editedSettings: editedSettingsSelector(state),
+        editing: editingSelector(state),
+        dockStyle: mapLayoutValuesSelector(state, {height: true})
     }), {
-        onClose: closeDetailsPanel
-    })(assign(require('../components/details/DetailsPanel'), {
+        onSave: save,
+        onEdit: edit.bind(null, 'content'),
+        onEditSettings: edit.bind(null, 'settings'),
+        onCancelEdit: cancelEdit,
+        onClose: setControlProperty.bind(null, 'details', 'enabled', false),
+        onSettingsChange: changeSetting,
+        onEditorUpdate: setEditedContent
+    }, (stateProps, dispatchProps, ownProps) => ({
+        ...stateProps,
+        ...ownProps,
+        ...dispatchProps,
+        editorProps: {
+            ...(stateProps.editorProps || {}),
+            ...(ownProps.editorProps || {}),
+            onUpdate: dispatchProps.onEditorUpdate
+        }
+    }))(Details),
+    containers: {
         BurgerMenu: {
             name: 'details',
             position: 1000,
             text: <Message msgId="details.title"/>,
             icon: <Glyphicon glyph="sheet"/>,
-            action: openDetailsPanel,
+            action: setControlProperty.bind(null, 'details', 'enabled', true),
+            selector: (state) => {
+                const mapId = mapIdSelector(state);
+                const detailsUri = mapId && mapInfoDetailsUriFromIdSelector(state, mapId);
+                const settings = settingsSelector(state) || {};
+                if (detailsUri && !settings.showAsModal) {
```

Comment:
Following my previous post
```suggestion
                if (detailsUri) {
```